# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: 33642b1c547b
# Full Hash: 33642b1c547b01c7a59417c8c2fe2a5ca33847c1
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 9: Implement alertclickcallback behavior r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227694
# ==============================================================================

diff -r 56ab49acff35 -r 33642b1c547b dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Thu Nov 07 17:32:06 2024 +0000
+++ b/dom/ipc/ContentParent.cpp	Thu Nov 07 17:32:07 2024 +0000
@@ -4751,20 +4751,12 @@
     return IPC_OK();
   }
 
-  if (aType.EqualsLiteral("click")) {
-    nsresult rv = swm->SendNotificationClickEvent(
-        aData.originSuffix(), aData.scope(), aData.ID(), aData.title(),
-        aData.dir(), aData.lang(), aData.body(), aData.tag(), aData.icon(),
-        aData.data(), aData.behavior());
-    Unused << NS_WARN_IF(NS_FAILED(rv));
-  } else {
-    MOZ_ASSERT(aType.EqualsLiteral("close"));
-    nsresult rv = swm->SendNotificationCloseEvent(
-        aData.originSuffix(), aData.scope(), aData.ID(), aData.title(),
-        aData.dir(), aData.lang(), aData.body(), aData.tag(), aData.icon(),
-        aData.data(), aData.behavior());
-    Unused << NS_WARN_IF(NS_FAILED(rv));
-  }
+  MOZ_ASSERT(aType.EqualsLiteral("close"));
+  nsresult rv = swm->SendNotificationCloseEvent(
+      aData.originSuffix(), aData.scope(), aData.ID(), aData.title(),
+      aData.dir(), aData.lang(), aData.body(), aData.tag(), aData.icon(),
+      aData.data(), aData.behavior());
+  Unused << NS_WARN_IF(NS_FAILED(rv));
 
   return IPC_OK();
 }